### ISSUE TITLE
unable to mmap 1024 bytes - Cannot allocate memory - even though there is more than enough ram

### ISSUE THREAD

**Issue by PeteKinz** [ThilinaRajapakse/simpletransformers#1157]
Hello everyone :)

I'm currently working on a seminar paper on nlp, summarization of sourcecode function documentation.
I've therefore created my own dataset with ca. 64000 samples (37453 is the size of the training dataset) and I want to fine tune the BART model:

![Screenshot from 2021-06-03 22-43-14](https://user-images.githubusercontent.com/85315971/120709441-24313600-c4bd-11eb-989c-a6f1427c5f3a.png)

![Screenshot from 2021-06-03 22-44-37](https://user-images.githubusercontent.com/85315971/120709585-4dea5d00-c4bd-11eb-982c-3ef007aa88c4.png)

everything fine so far BUT I get this error when I start the training.
![Screenshot from 2021-06-03 22-48-57](https://user-images.githubusercontent.com/85315971/120710045-f8fb1680-c4bd-11eb-8f18-f7a2ac4abc83.png)

One would think that I simply have not enough memory but this were my System Monitor ca. 3 sec. after the error:

![Screenshot from 2021-06-03 22-47-44](https://user-images.githubusercontent.com/85315971/120710244-33fd4a00-c4be-11eb-91c8-b0e069bd64a8.png)

and this was the lowest my available or free memory get in the time between starting the training and getting the error:

![Screenshot from 2021-06-03 22-48-30](https://user-images.githubusercontent.com/85315971/120712623-2a291600-c4c1-11eb-87f6-45d3eae6e9c9.png)

After a lot of tuning I found out that for some reason every thing works fine when I train the model only with a dataset of the size of max. 21000. I doesn't madder if I train the "base" version or the "large-cnn" version of the BART model. I just depends on size of my dataset. The error always occurs in the "Creating features from dataset file at cache_dir/" time.

So what have I already tried:
 - I added a lot of swap memory (as you can see in the screenshot of my System Monitor)
 - reduced  the numbers of workers to 1
 - I increased the hard- as well as the softmax of my systems open files limit (-n) to 86000
![Screenshot from 2021-06-03 23-02-00](https://user-images.githubusercontent.com/85315971/120711487-ba665b80-c4bf-11eb-9408-6e6e6ff4d6eb.png)

 - I also tried to train the model in a google colab notebook but I had the same issue;  if the dataset size gets over ca. 21000 the training fails. Even after I doubled the memory of my colab session but still keeping the datset size just a little bit over the 21000 limit.

**Desktop (please complete the following information):**
![Screenshot from 2021-06-03 23-08-37](https://user-images.githubusercontent.com/85315971/120712220-aa02b080-c4c0-11eb-88ba-22d4fbcc2933.png)

After trying to solve this by myself for literally days I would me more than happy if anyone of you guys have an idea how I can solve this :)

**Comment by rajgar114** [comment]
Hey @PeteKinz Have you got any solution to this problem ?

**Comment by PeteKinz** [comment]
Unfortunately not.
I played around with the `torch.nn.parallel.DistributedDataParallel` class, but this also couldn't solve my problem...

**Comment by PeteKinz** [comment]
One thing I noticed is that I have one package conflict in my colab notebook:
`ERROR: google-colab 1.0.0 has requirement ipykernel~=4.10, but you'll have ipykernel 5.5.5 which is incompatible.`
`ERROR: pydeck 0.6.2 has requirement ipykernel>=5.1.2; python_version >= "3.4", but you'll have ipykernel 4.10.1 which is incompatible.`

But the behavior of my notebook is exactly the same with either the 5.5.5 or the 4.10.1 ipykernel version.
And I don't have this conflict on my local machine, but still get the 'Cannot allocate memory (12)' error from above.

**Comment by rajgar114** [comment]
@PeteKinz 
Set the `use_multiprocessing` parameter of model arguments as False and it will work fine.

`model_args.use_multiprocessing = False`

**Comment by PeteKinz** [comment]
@rajgar114 
This fixed my problem!
Thank you very much. This was so obvious, but I didn't thought about that.
